### PR TITLE
Landing page

### PR DESCRIPTION
**Pull request by johanna-skylight**
## Description
This work implements the landing page with placeholder content. Changes include:

- Refactor of index.scss. Now only two main scss files exists and are imported: 
  `uswds-customized.scss`: handles all integration with uswds. Imports, settings setup and overwrites to the styles.
   `global.scss`: styles that are specific to our application.
- Improvements to uswds integration: Configures color settings in `uwds-core`, adds `browserlistsrc` configuration, removal of redundant sass variables.
- Creates `header` and `footer` components and integrates them in the root layout.
- Implements the landing page (styles and images added as well).
- Disabled image optimization in the nextjs configuration as the application is not run with a nextjs server.
-  Routed for static files now mounted at the root because images need to be served at the same level as pages. Order to setup the web application routes changed to page routes setup first and static assets routes last to have more specific to less specific order. 

**General Setup**
- Updates gitflow files to add PRs to demo in the slack notifications.
- Adds classnames dependency. This will allow us to breakup the setting of classes in the HTML elements into multiple lines, helps with nice formatting.
- Adds lefthook.yml configuration to run pre-commit hook in local. Will run ruff fixes in *.py files and eslint and prettier fixes in *.ts, *.tsx, *.css files. If you don't have lefthook installed this file will take no effect.

## Related Issues
Implements #267 

## Additional Notes
See screenshot on how the page looks. Ignore the Next icon, it just shows because the application was running in dev mode when I took it!.

landing_page

<--------------------- REMOVE THE LINES BELOW BEFORE MERGING --------------------->

## Checklist
Please review and complete the following checklist before submitting your pull request:

- [ ] I have ensured that the pull request is of a manageable size, allowing it to be reviewed within a single session.
- [ ] I have reviewed my changes to ensure they are clear, concise, and well-documented.
- [ ] I have updated the documentation, if applicable.
- [ ] I have added or updated test cases to cover my changes, if applicable.
- [ ] I have minimized the number of reviewers to include only those essential for the review.

## Checklist for Reviewers
Please review and complete the following checklist during the review process:

- [ ] The code follows best practices and conventions.
- [ ] The changes implement the desired functionality or fix the reported issue.
- [ ] The tests cover the new changes and pass successfully.
- [ ] Any potential edge cases or error scenarios have been considered.

### CODE REVIEW

File: src/api/recordlinker/main.py [hunk 1]
```diff
@@ -67,13 +67,6 @@
 app.mount("/api", api)
 
 if settings.ui_static_dir:
-    # static files for the UI
```
[the remark this answers is not in the document]
Comment (johanna-skylight) [reply src/api/recordlinker/main.py:70]: This needs to be the case for both local development and any lower(dev) and upper environment (prod). This is specific to the nextJS framework. Pages expect that the images they reference are served at the same level as them, so images need to be served at '/' in case of our setup.

I didn't want to have to list each image individually, as we are doing with the pages so I just mounted the nextJS output folder at `/` and didn't enable serving html (`html = false` by default).

The pages still need to be setup individually. And the previous `_next` folder, which contains js and css bundles, is still being resolved under the same path `/_next`.

Screenshot 2025-04-14 at 1 48 36 PM

[the remark this answers is not in the document]
Comment (johanna-skylight) [reply src/api/recordlinker/main.py:70]: I am happy to jump on a huddle if needed.

**Review by m-goggins — APPROVED** [review]
This looks fantastic to me! And I confirmed that I was able to get it up and running on my machine after tweaking the commands little for powershell. Thanks for all of your hard work on this!

File: src/ui/package.json [hunk 2]
```diff
@@ -14,6 +14,7 @@
   "dependencies": {
     "@trussworks/react-uswds": "^9.1.0",
     "@uswds/uswds": "^3.12.0",
+    "classnames": "^2.5.1",
```
[the remark this answers is not in the document]
Comment (johanna-skylight) [reply src/ui/package.json:17]: I had conflicts in package-lock so I regenerated it. It should be good.

File: lefthook.yml [hunk 3]
```diff
@@ -0,0 +1,17 @@
+pre-commit:
+  parallel: true
+  commands:
+    tests-code-checks:
```
[the remark this answers is not in the document]
Comment (johanna-skylight) [reply lefthook.yml:4]: I installed in my mac with brew. Do I just need to list it after line 39?

File: pyproject.toml [hunk 4]
```diff
@@ -36,7 +36,8 @@ dependencies = [
     "cryptography",  # MySQL & MariaDB dependency
     "pyodbc", # Microsoft SQL Server
     # Normalization
-    "phonenumbers"
+    "phonenumbers",
+    "lefthook"
```
Comment (ericbuckley) [inline pyproject.toml:40]: @johanna-skylight would you mind putting this after line 61 instead.  This is only necessary for developers and we won't need to install this in the upcoming AWS env running the demo.

File: src/api/recordlinker/main.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (johanna-skylight) [reply src/api/recordlinker/main.py:70]: images route has been added

File: src/api/recordlinker/main.py [hunk 5]
```diff
@@ -83,13 +76,27 @@ async def not_found_handler(request, exc):
             return FileResponse(os.path.join(settings.ui_static_dir, "404.html"), status_code=404)
         raise exc
 
+    # page routes
     @app.get("/")
     @app.get("/wizard")
-    @app.get("/favicon.ico")
```
Comment (ericbuckley) [inline src/api/recordlinker/main.py:88]: @johanna-skylight I think we still need to serve the favicon.ico from the root directory as thats where most browsers will look for it.

File: src/api/recordlinker/main.py [hunk 6]
```diff
@@ -83,13 +76,20 @@ async def not_found_handler(request, exc):
...
         view: str = path if path.endswith(".ico") else f"{path}.html"
         return FileResponse(os.path.join(settings.ui_static_dir, view))
+
+    # Other static files of NextJS app (includes index.html)
+    # [page-rehydration].txt files, images and js/css bundles in _next
+    app.mount(
+        "/",
+        StaticFiles(directory=os.path.join(settings.ui_static_dir), html = True),
+        name="SpaStatic",
+    )
```
Comment (johanna-skylight) [inline src/api/recordlinker/main.py:95]: @ericbuckley I noticed that the path to the configuration files that instruct how to do page hydration (This is what allows the multiple pages to behave as SPA after first load) was broken.

I have mounted the root of the nextJS output to `/` but this time I have configured the index.html to be served as part of the same mount configuration, I hope this alleviates the concern of the same path being setup in multiple ways.

Comment (ericbuckley) [reply src/api/recordlinker/main.py:95]: @johanna-skylight are those the index.txt and wizard.txt files?

My biggest concern was that we have multiple routes that will match the same URL pattern (eg `/wizard` matches both the routes on line 80 and 91) and is there any chance that fastapi will randomly select 1 over the other when that happens.  From what I can tell it's not random, and fastapi is using the route that was registered first, so I don't want to hold this up.

I'm going to add an issue for me to investigate this a little more in the future, but I believe this is something we can come back to.

**Review by ericbuckley — APPROVED** [review]

File: src/api/recordlinker/main.py (hunk 6 above)
Comment (johanna-skylight) [reply src/api/recordlinker/main.py:95]: Yes, I am referring to the index.txt and wizard.txt. Regarding `multiple routes that will match the same URL pattern (eg /wizard)`, the app.mount in line 91 does not resolve html pages by default. The only page that it resolves when setting `http = True` is `/` to `index.html`. It does not resolved any other paths to their html equivalents. 

You can test this in local. If you comment out the wizard route you will see the page will no longer load. This is why I started defining the page routes independently. I hope this clears the last concern.
